beetsplug/spotify.py

```diff
@@ -189,6 +189,9 @@ def _handle_response(self, request_type, url, params=None):
                     seconds.', seconds)
                 time.sleep(int(seconds) + 1)
                 return self._handle_response(request_type, url, params=params)
+            elif 'analysis not found' in response.text:
```

Comment:
Would it be possible to check the status code instead of looking at the text? For example, `response.status_code == 404`.

Reply:
I'm wondering if there may be other 404, which we have not encountered yet. As I test this more extensively over time, we can change this to 404 if required. Right now, my focus was to capture all the errors that we can encounter. Hope that makes sense. Let me know if you still prefer to change it to 404. I'll fix the other issues soon.

Reply:
Ah, I think I see now, and I was a bit too hasty. Namely, this is in `_handle_response`, which is used for all sorts of API calls, not just for querying the audio analysis. If it were the latter, then I think it would be safe to assume "404" meant "no audio analysis found."

With that in mind, it's a teensy bit troublesome that there is endpoint-specific logic in this general API wrapper function. It now means that, _technically_, the `_handle_response` function _could_ return `None` for any of its calls in this plugin. I wonder if we shouldn't instead just do `response.raise_for_status()` and handle the exception in the audio-features-specific code?

Reply:
Moved it to `track_audio_features` function. Can you check there...thanks!

Reply:
Looking like this is going in the right direction! But since this is raising a generic `SpotifyAPIError`, it seems like this condition should now just be `response.status_code == 404` instead of looking for this particular substring in the response? That way, different uses of `_handle_response` can still handle 404s in different ways.